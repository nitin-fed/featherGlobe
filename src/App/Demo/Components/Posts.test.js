import React from 'react';
import renderer from 'react-test-renderer';
import { Posts } from './Posts';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const initialState = {
  postReducer: {
    posts: [],
    postId: 0,
    isAddPostVisible: false,
    isEditing: false,
    showBackdrop: false,
    selectedPost: {}
  }
};
const mockStore = configureStore([]);
let store;

store = mockStore(initialState);

describe('Posts', () => {
  let props = {};
  beforeEach(() => {
    props = {
      onFetchPosts: jest.fn(),
      posts: [
        {
          body: 'test post body',
          id: 1,
          title: 'test post title',
          userId: 1
        },
        {
          body: 'test post body',
          id: 2,
          title: 'test post title',
          userId: 2
        }
      ],
      onLoadPostDescription: jest.fn(),
      show: true,
      onDeletePost: jest.fn(),
      onAddPost: jest.fn()
    };
  });

  const renderWithStore = p => {
    return renderer.create(
      <Provider store={store}>
        <Posts {...p} />
      </Provider>
    );
  };

  test('Should render if posts===obj', () => {
    const component = renderWithStore(props);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render if posts===undefined', () => {
    const newProps = {
      ...props,
      posts: undefined
    };
    const component = renderWithStore(newProps);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  xit('button click', () => {
    const component = renderWithStore(props);
    const tree = component.toJSON();
    const myButton = tree.children.find(item => {
      return item.type === 'button';
    });
    console.log(myButton.props.onClick);
    myButton.props.onClick({
      target: { value: 'blue' }
    });
    expect(onAddPost).toHaveBeenCalledWith('blue');
  });
});
