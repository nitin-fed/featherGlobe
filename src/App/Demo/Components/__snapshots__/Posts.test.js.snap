// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Should render if posts===obj 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Add Post
  </button>
  <div>
    <div
      className="Backdrop"
      onClick={[Function]}
    />
    <div
      className="postDescription"
      style={
        Object {
          "top": "200px",
        }
      }
    >
      <h1 />
      <h3 />
      <p />
      <button>
        Delete
      </button>
    </div>
  </div>
  <ul
    className="container main leftAlign posts"
  >
    <li>
      <div
        className="row"
      >
        <div
          className="col-md-1"
        >
          1
        </div>
        <div
          className="col-md-10"
          onClick={[Function]}
        >
          test post title
        </div>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </li>
    <li>
      <div
        className="row"
      >
        <div
          className="col-md-1"
        >
          2
        </div>
        <div
          className="col-md-10"
          onClick={[Function]}
        >
          test post title
        </div>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Posts Should render if posts===undefined 1`] = `
<h1>
  No Posts Found.
</h1>
`;
